app/controllers/articles_controller.rb

```diff
@@ -22,6 +22,15 @@ def destroy
     redirect_to root_path
   end
 
+  def update
+    @article = Article.find_by(id: params[:id])
+    @article.update(article_params) ? (redirect_to article_path(@article)) : (render 'edit')
+  end
+
+  def edit
+    @article = Article.find_by(id: params[:id])
```

Comment:
About setting the `@article` variable. You are using the same code in your `update` and `edit` methods. please consider extracting to a separate method and calling it in  a `before_action`